### ISSUE TITLE
Multi-select of consecutively listed facet objects

### ISSUE THREAD

**Issue by jquartel** [OpenRefine/OpenRefine#6337]
I sometimes find I need to select ('include') more than a handful of consecutive objects from a text facet and since clicking each one in the facet takes a while, I find the only way to do this to to make another boolean-output facet that discriminates the set I need. This works fine, but it occurred to me that it would be nice to be able to multi-select the facet objects with a shift-click as is common for other listings (like file browsers).

For example, suppose I have a text column 'Name' with a text facet and there are many unique names. The facet, by default, lists the names alphabetically, though you can switch to order by count. If I want to select/include all of the names starting with A plus the first few starting with B then, rather than clicking my way through each name (or coming up with another boolean facet), the proposed feature would allow me to just select the first A name, then scroll down to the last B name I want included and shift-click on it. This then selects all of the options between the first and last items I clicked (inclusively). If the facet were ordered by count, this feature would allow me to easily select, say, the 20 most common names.

**Comment by thadguidry** [comment]
So there would be a range.  And the start of the range is indicated how?
If I have a set of 5 and I have some checked as below, is there no user interaction needed to indicate the start of the range?

[X] 1
[ ] 2
[X] 3
[ ] 4
[ ] 5

Given the above state of a Facet, what happens if I just SHIFT+LMB click on 5?
Would only 4 also be checked automatically to include in the range?

I do understand you want a user interaction with the SHIFT+LMB click to choose the end of the range.

**Comment by jquartel** [comment]
The start of the range is given by an already-selected object, so yes it needs further clarification when there are more than one. I've just looked to see what Mac's file browser does and it's actually a bit complicated since it depends on the most recently selected object. This would probably be a step too far for OR, but I think the following simple logic would suffice for a shift+LMB click on a facet object:

case, there are no selected objects : select the clicked object as as though it were a normal click
case, the clicked object is already selected : do nothing
case, there is one selected object : select all object listed between and including that object and the clicked object
case, there are multiple selected objects AND all selected objects are before the clicked object : select all objects after the highest selected one, up to and including the clicked object
case, there are multiple selected objects AND all selected objects are after the clicked object : select all objects before the lowest selected one, down to and including the clicked object
case, there are multiple selected objects AND the clicked object falls between selected objects : just select the clicked object as though it were a normal click

**Comment by tfmorris** [comment]
It seems to me that anything which doesn't match the behavior of the host UI is going to be confusing to the user. We should just use the standard system UI conventions (and implementation, if possible).

**Comment by jquartel** [comment]
Agreed. I just don't know if the code currently supports remembering which is the most recently selected object, so I proposed what I think would still be logical behaviour without requiring this (and still satisfying the requirements for the original case descriptions) .
